### DISCUSSION TITLE
Proxying traffic on a real device iOS

### DISCUSSION THREAD

**Discussion by SergeyKrivorotko1** [mockoon/mockoon#1439] [Help]
Good afternoon.
How to proxy traffic on a real device (iPhone)?
On the device I use a proxy through the assembly machine (the device and the assembly machine are connected to the same point on the Internet)

**Comment by 255kb** [comment]
Hi, I'm not familiar with iOS development, but can't you point your app to call Mockoon running on your local machine? 

For what it's worth, Mockoon cannot act as a regular proxy. The [proxy feature](https://mockoon.com/docs/latest/server-configuration/proxy-mode/) allows you to forward calls to an existing API only.

**Reply by SergeyKrivorotko1** [reply]
This works for me locally when running tests on simulators, but I need to run tests on real devices. I set up a proxy on the device, but mockoon didn't work, maybe I need to install a certificate on the device, which I don't haveThis works for me locally when running tests on simulators, but I need to run tests on real devices. I set up a proxy on the device, but mockoon didn't work, maybe I need to install a certificate on the device, which I don't have

**Comment by SergeyKrivorotko1** [comment]
Hello
When using traffic sniffers such as Charles or Proxyman, it is possible to install a certificate on the device and trust it, specify the IP for the network used and sniff traffic. Is there a similar option with mockoon?

**Reply by 255kb** [reply]
Hi, no there is no such option yet.

**Reply by SergeyKrivorotko1** [reply]
Hi, thanks for the answer, is there a plan to add this option?

**Reply by 255kb** [reply]
No. But I'm not sure this would be useful as Mockoon cannot act as a real proxy like proxyman or Charles. It's rather a software you put in the middle between your app and live API, which shouldn't require to add self-signed certificates to your device.
